models/migrations/v176.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2021 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package migrations
+
+import (
+	"fmt"
+
+	"xorm.io/builder"
+	"xorm.io/xorm"
+)
+
+func deleteOrphanedIssueLabels(x *xorm.Engine) error {
+	type IssueLabel struct {
+		ID      int64 `xorm:"pk autoincr"`
+		IssueID int64 `xorm:"UNIQUE(s)"`
+		LabelID int64 `xorm:"UNIQUE(s)"`
+	}
+
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	if err := sess.Sync2(new(IssueLabel)); err != nil {
+		return fmt.Errorf("Sync2: %v", err)
+	}
+
+	if _, err := sess.In("id", builder.Select("issue_label.id").From("issue_label").
+		Join("LEFT", "label", "issue_label.label_id = label.id").
```

Comment:
Mysql is not going to like this:

            	            	migrate: do migrate: Error 1093: You can't specify target table 'issue_label' for update in FROM clause

The reference to issue_label needs to have an alias. 

This is exactly the same problem I faced in my PRs around issue_label

Reply:
the stupid thing is: "it works on my pc" ... just fine :/